### PR TITLE
wait_until: Add type annotations

### PR DESCRIPTION
**Pull request by drothlis**
TODO:

- [x] Test autocompletion in VS Code.

### CODE REVIEW

**Comment by drothlis** [comment]
It works! Before:

vscode before

After:

vscode after

File: _stbt/wait.py [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 
 
-def wait_until(callable_, timeout_secs=10, interval_secs=0, predicate=None,
-               stable_secs=0):
+T = TypeVar("T")
+
+
+def wait_until(callable_: Callable[[], T],
+               timeout_secs: int = 10,
+               interval_secs: int = 0,
```
Comment (wmanley) [inline _stbt/wait.py:15]: These should be `float`

```suggestion
               timeout_secs: float = 10,
               interval_secs: float = 0,
```

File: _stbt/wait.py [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
+               predicate: Callable[[T], bool] = None,
+               stable_secs: int = 0) -> T | None:
```
Comment (wmanley) [inline _stbt/wait.py:17]:
```suggestion
               stable_secs: float = 0) -> T | None:
```

**Review by wmanley — CHANGES_REQUESTED** [review]

**Comment by wmanley** [comment]
Very helpful.  I had a branch where I'd annotated all of our types at some point, I'll have to find it.

File: _stbt/wait.py (hunk 2 above)
Comment (drothlis) [reply _stbt/wait.py]: Done.

File: _stbt/wait.py (hunk 1 above)
Comment (drothlis) [reply _stbt/wait.py]: Done.

**Review by wmanley — APPROVED** [review]

**Pull request merged**